src/Data/Repositories/WalletRepository.cs

```diff
@@ -134,18 +136,54 @@ public async Task<(bool, string?)> AddRangeAsync(List<Wallet> type)
             return _repository.RemoveRange(types, applicationDbContext);
         }
 
-        public (bool, string?) Update(Wallet type, bool includeKeysUpdate = false)
-        {
+        public async Task<(bool, string?)> Update(Wallet type)
+        { 
             using var applicationDbContext = _dbContextFactory.CreateDbContext();
+            var wallet = await applicationDbContext.Wallets.Include(w => w.Keys).FirstOrDefaultAsync(x => x.Id == type.Id);
 
             type.SetUpdateDatetime();
-            if (!includeKeysUpdate)
-                type.Keys?.Clear();
 
-            type.ChannelOperationRequestsAsSource?.Clear();
-            return _repository.Update(type, applicationDbContext);
+            bool wasHotWallet = wallet.IsHotWallet;
+            applicationDbContext.Entry(wallet).CurrentValues.SetValues(type);
+            
+            if (!wasHotWallet && type.IsHotWallet)
+            {
+                var userKeys = wallet.Keys.Where(k => !string.IsNullOrEmpty(k.UserId));
+                foreach (var key in userKeys)
+                {
+                    wallet.Keys.Remove(key);
+                }
+            } 
+            else
+            {
+                foreach (var key in type.Keys)
+                {
+                    if (!Key.Contains(wallet.Keys, key))
+                    {
+                        wallet.Keys.Add(key);
+                    }
+                }
+            }
+            
```

Comment:
This is related to the keys update? right?

Reply:
exactly. If the user edits the wallet before finalising it and it makes it hot, we want to remove the user keys added.